webapp/app/js/perf.js

```diff
@@ -310,6 +334,40 @@ perf.controller('PerfCtrl', [ '$state', '$stateParams', '$scope', '$rootScope',
       plotGraph();
     }
 
+    $scope.resetHighlight = function() {
+      $scope.seriesList.forEach(function(series) {
+        series.highlighted = [];
+      });
+      $scope.revisionToHighlight = "";
+      $scope.resetHighlightButton = false;
+      updateURL();
+      plotGraph();
+    }
+
+    $scope.highlightRevision = function() {
+      var rev = $scope.revisionToHighlight;
+      if (rev.length == 12) {
+        var plotdata = $scope.plot.getData();
+        plotdata.forEach(function(series, i) {
```

Comment:
Instead of doing this several times (one for each series), why don't we call map on a set of $http promises, and then call plotGraph once at the end. You can see an example of this here: https://github.com/mozilla/treeherder-ui/blob/master/webapp/app/js/perf.js#L238 Also, unless I'm missing something, we should be able to call updateURL just once, at the beginning. 

i.e. 
    updateURL();
    $q.all($scope.plot.getData().map(function(series) { 
      if (series.points.show) { return $http.get(...).then(function(response) { ... }); }
      return null; 
     }));